src/Data/List/Relation/Binary/Permutation/Setoid/Properties.agda

```diff
@@ -454,52 +398,102 @@ module _ {ℓ} {R : Rel A ℓ} (R? : B.Decidable R) where
+-- But for the properties of steps which require precise measurement
+
+↭-respʳ-≋ : _↭_ Respectsʳ _≋_
+↭-respʳ-≋ xs≋ys               (refl zs≋xs)         = refl (≋-trans zs≋xs xs≋ys)
+↭-respʳ-≋ (x≈y ∷ xs≋ys)       (prep eq zs↭xs)      = prep (≈-trans eq x≈y) (↭-respʳ-≋ xs≋ys zs↭xs)
+↭-respʳ-≋ (x≈y ∷ w≈z ∷ xs≋ys) (swap eq₁ eq₂ zs↭xs) = swap (≈-trans eq₁ w≈z) (≈-trans eq₂ x≈y) (↭-respʳ-≋ xs≋ys zs↭xs)
+↭-respʳ-≋ xs≋ys               (trans ws↭zs zs↭xs)  = trans ws↭zs (↭-respʳ-≋ xs≋ys zs↭xs)
+
+↭-respˡ-≋ : _↭_ Respectsˡ _≋_
+↭-respˡ-≋ xs≋ys               (refl ys≋zs)         = refl (≋-trans (≋-sym xs≋ys) ys≋zs)
+↭-respˡ-≋ (x≈y ∷ xs≋ys)       (prep eq zs↭xs)      = prep (≈-trans (≈-sym x≈y) eq) (↭-respˡ-≋ xs≋ys zs↭xs)
+↭-respˡ-≋ (x≈y ∷ w≈z ∷ xs≋ys) (swap eq₁ eq₂ zs↭xs) = swap (≈-trans (≈-sym x≈y) eq₁) (≈-trans (≈-sym w≈z) eq₂) (↭-respˡ-≋ xs≋ys zs↭xs)
+↭-respˡ-≋ xs≋ys               (trans ws↭zs zs↭xs)  = trans (↭-respˡ-≋ xs≋ys ws↭zs) zs↭xs
+
+-- Properties of steps using the above
+
+0<steps : (xs↭ys : xs ↭ ys) → 0 < steps xs↭ys
+0<steps (refl _)             = z<s
+0<steps (prep eq xs↭ys)      = m<n⇒m<1+n (0<steps xs↭ys)
+0<steps (swap eq₁ eq₂ xs↭ys) = m<n⇒m<1+n (0<steps xs↭ys)
+0<steps (trans xs↭ys xs↭ys₁) =
+  <-≤-trans (0<steps xs↭ys) (m≤m+n (steps xs↭ys) (steps xs↭ys₁))
+
+steps-respˡ : (ys≋xs : ys ≋ xs) (ys↭zs : ys ↭ zs) →
+              steps (↭-respˡ-≋ ys≋xs ys↭zs) ≡ steps ys↭zs
+steps-respˡ _               (refl _)            = refl
+steps-respˡ (_ ∷ ys≋xs)     (prep _ ys↭zs)      = cong suc (steps-respˡ ys≋xs ys↭zs)
+steps-respˡ (_ ∷ _ ∷ ys≋xs) (swap _ _ ys↭zs)    = cong suc (steps-respˡ ys≋xs ys↭zs)
+steps-respˡ ys≋xs           (trans ys↭ws ws↭zs) = cong (_+ steps ws↭zs) (steps-respˡ ys≋xs ys↭ws)
+
+steps-respʳ : (xs≋ys : xs ≋ ys) (zs↭xs : zs ↭ xs) →
+              steps (↭-respʳ-≋ xs≋ys zs↭xs) ≡ steps zs↭xs
+steps-respʳ _               (refl _)            = refl
+steps-respʳ (_ ∷ ys≋xs)     (prep _ ys↭zs)      = cong suc (steps-respʳ ys≋xs ys↭zs)
+steps-respʳ (_ ∷ _ ∷ ys≋xs) (swap _ _ ys↭zs)    = cong suc (steps-respʳ ys≋xs ys↭zs)
+steps-respʳ ys≋xs           (trans ys↭ws ws↭zs) = cong (steps ys↭ws +_) (steps-respʳ ys≋xs ws↭zs)
+
+------------------------------------------------------------------------
+-- DEPRECATED NAMES
+------------------------------------------------------------------------
+-- Please use the new names as continuing support for the old names is
+-- not guaranteed.
+
+-- Version 2.1
+
+split : ∀ (v : A) as bs {xs} → xs ↭ as ++ [ v ] ++ bs →
+        ∃₂ λ ps qs → xs ≋ ps ++ [ v ] ++ qs
+split v as bs xs↭as++[v]++bs
+  with ps , qs , eq , _ ← ↭-split v as bs xs↭as++[v]++bs
+  = ps , qs , eq
+{-# WARNING_ON_USAGE split
+"Warning: split was deprecated in v2.1.
+Please use the sharper lemma ↭-split instead."
+#-}
+
+foldr-commMonoid = foldr-pres-↭
+{-# WARNING_ON_USAGE foldr-commMonoid
+"Warning: foldr-commMonoid was deprecated in v2.1.
+Please use foldr-pres-↭ instead."
+#-}
```

Comment:
Premature, esp. in view of my comment elsewhere on preferring a (future!) version of this to take the *bundle* and not the *structure* as argument. But I should save the name for such downstream refactoring! Will now revert.